rest_framework/fields.py

```diff
@@ -1546,16 +1546,15 @@ def to_representation(self, value):
             return None
 
         use_url = getattr(self, 'use_url', api_settings.UPLOADED_FILES_USE_URL)
-
         if use_url:
-            if not getattr(value, 'url', None):
-                # If the file has not been saved it may not have a URL.
+            try:
+                url = value.url
+                request = self.context.get('request', None)
+                if request is not None:
+                    return request.build_absolute_uri(url)
+                return url
+            except AttributeError:
```

Comment:
Let's wrap the `AttributeError` *only* around the `url = value.url` line, right?

Reply:
OK